src/main/java/org/elasticsearch/search/internal/DefaultSearchContext.java

```diff
@@ -197,8 +195,27 @@ public DefaultSearchContext(long id, ShardSearchRequest request, SearchShardTarg
         this.indexShard = indexShard;
         this.indexService = indexService;
 
-        this.searcher = new ContextIndexSearcher(this, engineSearcher);
-
+        aliasFields = indexService.aliasesService().aliasFields(request.filteringAliases());
+        if (aliasFields != null) {
```

Comment:
can it also just be the empty set all the time. I really don't like null invariants